### PR TITLE
fix: export DB to as plain db during backup

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

1. backup files are encrypted
2. the passphrase is exported in the metadata file

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by vitorhugods** [comment]
Could you please fill the PR template?

Also, what about the restoring of the backup. I remember you had some concerns about it being able to be imported later using SQLCipher.

**Comment by MohamadJaara** [comment]
> Could you please fill the PR template?
> 
> Also, what about the restoring of the backup. I remember you had some concerns about it being able to be imported later using SQLCipher.

it does not work yet i will label it as WIP

**Comment by MohamadJaara** [comment]
> I remember you had some concerns about it being able to be imported later using SQLCipher. 

I did double-check, and we need to import the version manually https://www.zetetic.net/sqlcipher/sqlcipher-api/#notes-export

**Comment by MohamadJaara** [comment]
i did dig deeper into the issue with the sqlCipher_export not working in here
https://github.com/wireapp/kalium/pull/1322

i did find the reason in this interesting  thread
https://github.com/cashapp/sqldelight/discussions/2929
TLDR, for security reasons sqlCipher allow the export only by using a method rawExecSQL which is not exposed through  in sqlDelight (sqlDelight wraps SqlDatabase functions only)

so we have 2 options
1. copy the db as we do now, and change the key of the copy to something hard-coded (the rekey function does not allow empty passwords) .
pro: simple
con: we will still not be able to import a DB from prod app into a debug app

2.create a new database with the same schema and select -> insert data manually for each table (same as we do during import but reversed).
pro: cleaner solution and the backup DB can also have an independent schema from the local that is simpler and does not include some data that we are not restoring (for example clients table or metadata).
con: can take some time to implement.

**Comment by MohamadJaara** [comment]
is will close this PR in and open a new one targeting release
